### ISSUE TITLE
Fix `declaration-block-no-shorthand-property-overrides` false negatives for `border` after `border-image`

### ISSUE THREAD

**Issue by ybiquitous** [stylelint/stylelint#7586]
### What minimal example or steps are needed to reproduce the bug?

```css
a {
    border-image: linear-gradient(blue, red) 1; /* ignored */
    border: 5px solid red;
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "declaration-block-no-shorthand-property-overrides": true
  }
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilAjCATrAQLQCWAtkgOYzwoA2ZmMSp1BSUZMmALgApcDAK4wANCgIwoAShQBGANwoA9ACoUZapkIyU61Vhz4iMAvQCsABwAeKAM4QmUKTKVYAvlhDiQAMzIGGAA5JAo4RBhbcOtggDowBwdfcAhMQOoEEAxsFHQQAhFghwL6XJwcAtgwBjYkPjJ0kmEIMABrEl0SBwALQj5epEwoEmsCCGtzPgBPEggAN3MCMlhShBQ+IphjFG9MT1TIDO0AMUIqPmyAKydMVNhrFMQK-JAHWeCmfjK3hQA2eIAZniCgKXhAniAA)

Also, you can see the ignored `border-image` in [CodePen](https://codepen.io/ybiquitous/pen/eYoGaPe).

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "16.3.1"
}
```

### What did you expect to happen?

The following problem should be reported:

```css
a {
    border-image: linear-gradient(blue, red) 1; /* ignored */
    border: 5px solid red;
 /* ^^^^^^ Unexpected shorthand "border" after "border-image" */
}
```

### What actually happened?

No problems to be reported.

### Do you have a proposal to fix the bug?

Instead of adding `border-image` to the `border` Map in `reference.longhandSubPropertiesOfShorthandProperties`, custom logic should be implemented in `rules/declaration-block-no-shorthand-property-overrides/index.mjs` because `border-image` is just reset by `border`, not a longhand property. See also the [CSS spec](https://drafts.csswg.org/css-backgrounds/#propdef-border).

Background: https://github.com/stylelint/stylelint/pull/7585#discussion_r1545345582

**Comment by Mouvedia** [comment]
Should we fix `grid` at the same time?
see https://github.com/w3c/csswg-drafts/issues/1036#issuecomment-279699823

i.e. grid resets `grid-column-gap`/`column-gap` and `grid-row-gap`/`grid-row-gap`

**Comment by ybiquitous** [comment]
@Mouvedia Thanks for the info. But it seems `grid-gap` is not reset by the following `grid`. My example may be wrong, right? See my [CodePen](https://codepen.io/ybiquitous/pen/WNWXGzP).

```css
a {
  display: grid;
  grid-gap: 20px; /* unignored? */
  grid: 50px 50px / 100px 100px;
}
```

---

I suggest creating an issue for each property (e.g. `grid`, `font` etc.) so that we can minimize each PR. We can also write a PR to fix them together if we wish.

**Comment by Mouvedia** [comment]
> But it seems `grid-gap` is not reset by the following `grid`.

I didn't say `grid-gap`.

cf
> …`grid-column-gap`/`column-gap` and `grid-row-gap`/`row-gap`

> I suggest creating an issue for each property…

👍
